admin/class-boldgrid-backup-admin-db-import.php

```diff
@@ -166,8 +192,183 @@ public function import_lines( $lines ) {
+		$sql_security_offset = strpos( $line, 'SQL SECURITY' );
+		$line_fixed_definer  = substr( $line, 0, 9 );
+		if ( strpos( $line, '@`%`' ) ) {
+			$line_fixed_definer .= 'DEFINER=`' . DB_USER . '`@`%` ';
+		} else {
+			$line_fixed_definer .= 'DEFINER=`' . DB_USER . '`@`' . DB_HOST . '` ';
+		}
+
+		if ( strpos( $line, 'SQL SECURITY' ) ) {
+			$line_fixed_definer .= subStr( $line, $sql_security_offset );
+		} else {
+			$line_fixed_definer .= '*/';
+		}
+
+		return $line_fixed_definer;
+	}
+
+	/**
+	 * Tests if database user has specific privileges
+	 *
+	 * @since SINCEVERSION
+	 *
+	 * @param array $privileges An array of permissions to check against.
+	 * @return bool True if user has specified privileges.
+	 */
+	public function has_db_privileges( array $privileges ) {
+		$user_grants = $this->get_db_privileges();
+		if ( in_array( 'ALL', $user_grants, true ) ) {
+			return true;
+		}
+		if ( count( $privileges ) === count( array_intersect( $privileges, $user_grants ) ) ) {
+			return true;
+		}
+		return false;
+	}
+
+	/**
+	 * Get database user privileges
+	 *
+	 * @since SINCEVERSION
+	 *
+	 * @global wpdb $wpdb The WordPress database class object.
+	 *
+	 * @return array An array of database user privileges.
+	 */
+	public function get_db_privileges() {
+		$results = $this->show_grants_query();
+
+		foreach ( $results as $result ) {
+			$is_string_db_grant      = ( false !== strpos( $result[0], 'ON `' . DB_NAME . '`' ) );
+			$is_string_all_grant     = ( false !== strpos( $result[0], 'ON *.*' ) );
+			$is_grant_all_privileges = ( false !== strpos( $result[0], 'GRANT ALL PRIVILEGES' ) );
+
+			if ( ( $is_string_db_grant || $is_string_all_grant ) && $is_grant_all_privileges ) {
+				return [ 'ALL' ];
+			}
+			if ( ( $is_string_db_grant ) && false === $is_grant_all_privileges ) {
+				return $this->get_grants_array( $result[0] );
+			}
+		}
+		return [];
+	}
+
+	/**
+	 * Show Grants Query
+	 *
+	 * Queries the database for 'SHOW GRANTS'
+	 *
+	 * @since SINCEVERSION
+	 *
+	 * @return array an array of results from the database query
+	 */
+	public function show_grants_query() {
+		$db           = new PDO( sprintf( 'mysql:host=%1$s;dbname=%2$s;', DB_HOST, DB_NAME ), DB_USER, DB_PASSWORD );
+		$db_statement = $db->query( 'SHOW GRANTS' );
+		return $db_statement->fetchAll();
+	}
+
+	/**
+	 * Execute Import
+	 *
+	 * Executes Import MySql Query
+	 *
+	 * @since SINCEVERSION
+	 *
+	 * @param PDO $db The PDO Object.
+	 * @param string $sql_line The line of sql to execute.
+	 *
+	 * @return int Number of affected rows
+	 */
+	public function exec_import( PDO $db, $sql_line ) {
+		return $db->exec( $sql_line );
+	}
+
+	/**
+	 * Get a user's grants in the form of an array
+	 *
+	 * @since SINCEVERSION
+	 *
+	 * @param string $grants_string.
+	 * @return array An array of grants.
+	 */
+	public function get_grants_array( $grants_string ) {
+		if ( strpos( $grants_string, 'GRANT' ) === 0 ) {
+			$grants_string = substr( $grants_string, 6 );
```

Comment:
Use a comment to explain "magic number"; 6.

Reply:
Instead, I stored the value 6, in a variable $expected_grants_string_start, and used that.